src/Raven.Server/Rachis/FollowerAmbassador.cs

```diff
@@ -651,11 +659,49 @@ private unsafe void WriteSnapshotToFile(ClusterOperationContext context, Stream
                                                 var currentTreeValueKey = treeIterator.CurrentKey;
                                                 binaryWriter.Write(currentTreeValueKey.Size);
                                                 copier.Copy(currentTreeValueKey.Content.Ptr, currentTreeValueKey.Size);
-                                                var reader = treeIterator.CreateReaderForCurrent();
-                                                binaryWriter.Write(reader.Length);
-                                                copier.Copy(reader.Base, reader.Length);
-
-                                                totalSizeInBytes += sizeof(long) + currentTreeValueKey.Size + reader.Length;
+                                                totalSizeInBytes += sizeof(int) + currentTreeValueKey.Size;
+
+                                                switch (type)
+                                                {
+                                                    case TreeFlags.None:
+                                                    case TreeFlags.FixedSizeTrees:
+                                                    case TreeFlags.LeafsCompressed:
+                                                        var reader = treeIterator.CreateReaderForCurrent();
```

Comment:
If the current item is fixed size tree (the large one, not the embedded) then this will just return the header `FixedSizeTreeHeader.Large`, not the actual entries of that FST. I don't see where do we send them